src/huggingface_hub/templates/modelcard_template.md

```diff
@@ -81,15 +81,20 @@ Use the code below to get started with the model.
 
 {{ training_data | default("[More Information Needed]", true)}}
 
-## Training Procedure [optional]
+## Training Procedure 
 
 <!-- This relates heavily to the Technical Specifications. Content here should link to that section when it is relevant to the training procedure. -->
 
-### Preprocessing
+### Preprocessing [optional]
 
 {{ preprocessing | default("[More Information Needed]", true)}}
 
-### Speeds, Sizes, Times
+
+### Training hyperparameters
+- **Precision:** [More Information Needed] <!--fp16, bf16, fp8, fp32 -->
+- **Regime:** [More Information Needed] <!--mixed precision or not -->
```

Comment:
Does the above "Precision" not already tell you this (I really dont know)? Can you list examples of what you would put here?

Reply:
Not all.  You can do bf16 and bf16 mixed precision - 2 very different regimes.

please see https://huggingface.slack.com/archives/C025H22A3ND/p1675973358816029?thread_ts=1675839299.266509&cid=C025H22A3ND

My proposal was:

Either 2 separate entries:

1. Precision: fp16, bf16, fp8, fp32
2. Regime: mixed precision or not

Or one combined:

Training regime:

1. fp32
2. fp16 mixed precision
3. bf16 mixed precision
4. bf16 non-mixed precision
5. fp16 non-mixed precision
6. fp8 mixed precision

I think the combined is a much better option as it exactly removes the questions like you have posed above.

Reply:
if there was a way to do a pulldown list the 2nd option would be much easier for the user IMHO

Reply:
> if there was a way to do a pulldown list the 2nd option would be much easier for the user IMHO

In `huggingface_hub` it will most probably be a free-text field as the library is not opinionated on the content (it just renders a jinja template with some values provided as input). But if there is an UI somewhere for this, it would definitely makes sense.

(also related to @nateraw's [comment](https://github.com/huggingface/huggingface_hub/pull/1341#discussion_r1106323859) => it would be good to have these as variables)

Reply:
it could be in the metadata section if we want/need a more structured schema with a dropdown etc.

cc @osanseviero 

(spec for metadata is here: https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1)

Reply:
@julien-c All template kwargs get dumped to metadata so the information will be there on creation. :)
 https://github.com/huggingface/huggingface_hub/blob/ad314032248a48930d7c76c869627c980bf69b41/src/huggingface_hub/repocard.py#L312-L318